src/Core/Config/CoreConfigCreator.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace SilverStripe\Core\Config;
+
+use SilverStripe\Config\Collections\CachedConfigCollection;
+use SilverStripe\Config\Collections\MemoryConfigCollection;
+use SilverStripe\Config\Transformer\PrivateStaticTransformer;
+use SilverStripe\Config\Transformer\YamlTransformer;
+use SilverStripe\Control\Director;
+use SilverStripe\Core\Config\Middleware\ExtensionMiddleware;
+use SilverStripe\Core\Config\Middleware\InheritanceMiddleware;
+use SilverStripe\Core\Manifest\ClassLoader;
+use SilverStripe\Core\Manifest\ModuleLoader;
+use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * Factory for silverstripe configs
+ */
+class CoreConfigCreator
```

Comment:
What's the difference between a "factory" and a "creator"? We seem to use both terms in core